src/app/components/audios/audios.component.ts

```diff
@@ -1,9 +1,114 @@
+import { TableRowTemplateDirective } from 'src/app/directives/table-row-template.directive';
+import { LoadingComponent } from '@common/loading/loading.component';
+import { TabDirective } from 'src/app/directives/tab.directive';
+import { UserAudioRequestRow } from '@darts-types/index';
 
 @Component({
   selector: 'app-audios',
   templateUrl: './audios.component.html',
   styleUrls: ['./audios.component.scss'],
   standalone: true,
+  imports: [
+    DataTableComponent,
+    LoadingComponent,
+    TabsComponent,
+    CommonModule,
+    TableRowTemplateDirective,
+    RouterLink,
+    TabDirective,
+  ],
 })
-export class AudiosComponent {}
+export class AudiosComponent {
+  audioService = inject(AudioService);
+  datePipe = inject(DatePipe);
+
+  userId: number;
+
+  audioRequests$: Observable<UserAudioRequest[]>;
+  expiredAudioRequests$: Observable<UserAudioRequest[]>;
+
+  inProgessRows$: Observable<UserAudioRequestRow[]>;
+  completedRows$: Observable<UserAudioRequestRow[]>;
+  expiredRows$: Observable<UserAudioRequestRow[]>;
+
+  data$: Observable<{
+    inProgessRows: UserAudioRequestRow[];
+    completedRows: UserAudioRequestRow[];
+    expiredRows: UserAudioRequestRow[];
+  }>;
+
+  columns = [
+    { name: 'Case ID', prop: 'caseNumber', sortable: true },
+    { name: 'Court', prop: 'courthouse', sortable: true },
+    { name: 'Hearing date', prop: 'hearingDate', sortable: true },
+    { name: 'Start time', prop: 'startTime', sortable: true },
+    { name: 'End time', prop: 'endTime', sortable: true },
+    { name: 'Request ID', prop: 'requestId', sortable: true },
+    { name: 'Expiry date', prop: 'expiry', sortable: true },
+    { name: 'Status', prop: 'status', sortable: true },
+  ];
+
+  constructor(private route: ActivatedRoute) {
+    this.userId = this.route.snapshot.data.userState.userId;
+
+    this.audioRequests$ = this.audioService.getAudioRequestsForUser(this.userId, false);
+    this.expiredAudioRequests$ = this.audioService.getAudioRequestsForUser(this.userId, true);
+
+    this.inProgessRows$ = this.audioRequests$.pipe(
+      map((audioRequests) => this.filterInProgressRequests(audioRequests)),
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.completedRows$ = this.audioRequests$.pipe(
+      map((audioRequests) => this.filterCompletedRequests(audioRequests)),
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.expiredRows$ = this.expiredAudioRequests$.pipe(
+      map((audioRequests) => this.mapAudioRequestToRows(audioRequests))
+    );
+
+    this.data$ = combineLatest({
+      inProgessRows: this.inProgessRows$,
+      completedRows: this.completedRows$,
+      expiredRows: this.expiredRows$,
+    });
+  }
+
+  mapAudioRequestToRows(audioRequests: UserAudioRequest[]): UserAudioRequestRow[] {
+    return audioRequests.map((ar) => {
+      return {
+        // id: ar.case_id,
```

Comment:
Might need this for linking to the case screen?

Reply:
Again, awaiting the case_id to be added to the response.